sdk/core/azure-core/tests/testserver_tests/async_tests/test_rest_headers_async.py

```diff
@@ -7,9 +7,124 @@
 import pytest
 from azure.core.rest import HttpRequest
 
+# flask returns these response headers, which we don't really need for these following tests
+RESPONSE_HEADERS_TO_IGNORE = [
+    "Content-Type",
+    "Content-Length",
+    "Server",
+    "Date",
+]
+
+@pytest.fixture
+def get_response_headers(client):
+    async def _get_response_headers(request):
+        response = await client.send_request(request)
+        response.raise_for_status
+        for header in RESPONSE_HEADERS_TO_IGNORE:
+            response.headers.pop(header)
+        return response.headers
+    return _get_response_headers
+
+@pytest.mark.asyncio
+async def test_headers_response(get_response_headers):
+    h = await get_response_headers(HttpRequest("GET", "/headers/duplicate/numbers"))
+    assert "a" in h
+    assert "A" in h
+    assert "b" in h
+    assert "B" in h
+    assert "c" not in h
+    assert h["a"] == "123, 456"
+    assert h.get("a") == "123, 456"
+    assert h.get("nope", default=None) is None
+
+    assert list(h.keys()) == ["a", "b"]
+    assert list(h.values()) == ["123, 456", "789"]
+    assert list(h.items()) == [("a", "123, 456"), ("b", "789")]
+    assert list(h) == ["a", "b"]
+    assert dict(h) == {"a": "123, 456", "b": "789"}
+
+@pytest.mark.asyncio
+async def test_headers_response_keys(get_response_headers):
+    h = await get_response_headers(HttpRequest("GET", "/headers/duplicate/numbers"))
+    # basically want to make sure this behaves like dict {"a": "123, 456", "b": "789"}
+    ref_dict = {"a": "123, 456", "b": "789"}
+    assert list(h.keys()) == list(ref_dict.keys())
+    assert repr(h.keys()) == repr(ref_dict.keys())
+    assert "a" in h.keys()
+    assert "b" in h.keys()
+    assert set(h.keys()) == set(ref_dict.keys())
```

Comment:
For testing keys/values/items, we can also test that mutability is preserved:
```python
before_mutation_keys = h.keys()
h['c'] = '000'
assert 'c' in before_mutation_keys
```

Reply:
@annatisch I looked into this behavior, and httpx and aiohttp both don't natively have this behavior. Is this something we want to override from them?

Reply:
fixed!